test/test_sparse_csr.py

```diff
@@ -2516,12 +2516,24 @@ def _to_from_layout(layout_a, layout_b):
             # CSC to CSC conversion is supported
             if layout_a is torch.sparse_csc and layout_b is torch.sparse_csc:
                 expect_error = False
+            if layout_a is torch.sparse_bsc and layout_b is torch.sparse_csc:
+                expect_error = False
             if expect_error:
                 with self.assertRaises(RuntimeError):
                     b = self._convert_to_layout(a, layout_a)
                     self._convert_to_layout(b, layout_b)
             else:
-                b = self._convert_to_layout(a, layout_a)
+                if layout_a is torch.sparse_bsc:
+                    # workaround no dense<->bsc
+                    b_bsr = self._convert_to_layout(a.transpose(-2, -1).contiguous(), torch.sparse_bsr)
+                    # bsr of the transpose is bsc of this
+                    b = torch._sparse_bsc_tensor_unsafe(b_bsr.crow_indices(),
+                                                        b_bsr.col_indices(),
+                                                        b_bsr.values().transpose(-2, -1).contiguous(),
+                                                        size=a.shape)
+                    torch._validate_sparse_compressed_tensor_args(b.ccol_indices(), b.row_indices(), b.values(), b.shape, b.layout)
+                else:
+                    b = self._convert_to_layout(a, layout_a)
                 c = self._convert_to_layout(b, layout_b)
                 if (layout_a is not torch.sparse_bsr and layout_b is not torch.sparse_bsr):
```

Comment:
This entire test needs to be re-worked. The inner section here where we used `dense->transpose->bsr->bsc` is not needed any more since `dense->bsc` is enabled directly earlier in the stack. 

Also, this checks 2-d conversions only and I think there is a bug in the additions from this PR for some n-d cases which are not caught. 

Your confusion is not misplaced.